### ISSUE TITLE
Issue with alert and example colors

### ISSUE THREAD

**Issue by sgargbugreporter** [matze/mtheme#235]
I was trying to change the example color and ran into a few problems. 

- [mwe_title.txt](https://github.com/matze/mtheme/files/565750/mwe_title.txt) is a minimal working example (MWE) that outputs 2 slides - 1st one is a title and the second one is the slide of interest. This is simply slide 14 (Blocks) from [demo.tex](https://github.com/matze/mtheme/blob/master/demo/demo.tex). I am attempting to change the color of the `alerted` text and the `example` text. The color of the `alerted` text gets changed, but the `example` text _remains the same_. This seems to be a bug somewhere. See the output: [mwe_title.pdf](https://github.com/matze/mtheme/files/565745/mwe_title.pdf)

- [mwe_NOtitle.txt](https://github.com/matze/mtheme/files/565749/mwe_NOtitle.txt) is another minimal working example (MWE) where I have commeted out the `\maketitle` command so that it only outputs 1 slides - the `Blocks` slide.  In this case none of the colors gets changed, not even the `alerted` that was getting changed earlier. See the output: [mwe_NOtitle.pdf](https://github.com/matze/mtheme/files/565744/mwe_NOtitle.pdf)

So, to recap:
1. `example` text color does **not** change
1. If there is no `\maketitle` **no** color-changing occurs.

**Comment by rchurchley** [comment]
For a practical solution, moving the `\setbeamercolor` declarations to the preamble seem to fix both problems.

**Comment by sgargbugreporter** [comment]
Thank you. Although that does solve the problem, it begs a larger question - should the color declarations be in teh body (as in the demo.tex file) or in the preamble? As putting them in the preamble does solve these issues, IMHO the preamble is the right place. If so, then demo.tex and other examples should do that and put a warning about this.

**Comment by rchurchley** [comment]
It's my own habit to generally put declarations in the preamble. That said, you are right to report this behaviour as a bug; changing the colours mid-presentation *should* be possible, and we should keep this issue open until we figure out how to fix it.

Just thought I'd provide a quick work-around until I (or someone else) can find time to delve deeper into Beamer's colour management system and see why this is happening.

**Comment by benjamin-weiss** [comment]
It is weird. The beamer colors `alerted text` and `example text` need to be used at least once, that the derived colors get updated. So this seems to work fine:
```TeX
\setbeamercolor{alerted text}{fg=CardinalRed}
\setbeamercolor{example text}{fg=Tangerine}
{
  \usebeamercolor[fg]{alerted text}
  \usebeamercolor[fg]{example text}
}
...
```
Looks like a beamer bug.
